src/modules/sipcapture/sipcapture.c

```diff
@@ -434,6 +436,64 @@ struct module_exports exports = {
 	destroy          /* module destroy function */
 };
 
+/*
+ * Find the first occurrence of find in s, where the search is limited to the
+ * first slen characters of s.
+ */
+char *strnstr(const char *s, const char *find, size_t slen)
+{
+	char c, sc;
+	size_t len;
+
+	if ((c = *find++) != '\0') {
+		len = strlen(find);
+		do {
+			do {
+				if ((sc = *s++) == '\0' || slen-- < 1)
+					return (NULL);
+			} while (sc != c);
+			if (len > slen)
+				return (NULL);
+		} while (strncmp(s, find, len) != 0);
+		s--;
+	}
+
+	return ((char *)s);
+}
+
+int force_capture_callid (struct sip_msg *msg, struct _sipcapture_object *sco)
+{
+	char *tmp = NULL;
+	char *end;
+	struct hdr_field *hdr = NULL;
+
+	tmp= msg->unparsed;
+	end = msg->buf+msg->len;
+	tmp = strnstr(tmp, "Call-ID", (int)(end - tmp) );
+
+	if (tmp == NULL) {
+		LM_DBG("Bad msg callid not found\n");
+		EMPTY_STR(sco->callid);
+	} else {
+		hdr=pkg_malloc(sizeof(struct hdr_field));
+		if (unlikely(hdr==0)){
+			LOG(L_ERR, "ERROR:parse_headers: memory allocation error\n");
```

Comment:
consider removing this log statement and just use PKG_MEM_ERROR macro instead. The "parse_headers" prefix makes also not much sense in this context, I think.